### ISSUE TITLE
500 Server Error Changing Cardype from File or Image to Anything Else

### ISSUE THREAD

**Issue by bertdeb** [decko-commons/decko#619]
Brand new deck, create new card `newcard` and set its **Cardtype** to `File`. Save.

Now change Cardtype to `RichText`:

```
Wild Card!
500 Server Error
Yuck, sorry about that.
```

Same error when attempting to change the **Cardtype** to anything else.

Similarly, create `newcard`, set its **Cardtype** to `Image`, submit an image file and save.

Changing the **Cardtype** to anything else also results in the `500 Server Error`.

If neither the `File` or `Image` Cardtypes can be changed, maybe the option to change it should be removed? Or a warning that the image/file will be unlinked / removed?

**Comment by ethn** [comment]
Nice catch.

tl;dr - I'll fix the bug; it's pretty shallow. In the meantime, if you need to change a type like this, you can get there using ?view=edit_cardtype.

Recently we upgraded the UI so that you could see (and immediately fix) the content impacts of a type change. But that UI is not accounting for type changes in which the ui for the type you're changing _from_ sends attributes that the type you're changing _to_ can't handle. 

The issue, in effect, is that the UI is sending a bad request (and one that our tests are not covering).  There is no underlying reason why the card can't be changed. The underlying file/image will not be lost, but you're right that we should explain the impacts for existing links to that file/image and what will need to happen to use it again.
